### PR TITLE
add support for https proxy

### LINKED ISSUE #31113: HTTPS Proxy support feature requests
Hello,
As the support of HTTPS proxies is becoming more common, we need to execute HTTP and HTTPS requests through an HTTPS proxy.

To be clear, this is not about executing an HTTPS request through an HTTP proxy (which works well), it is about connecting to the proxy itself using HTTPS.

As an additional requirement, we are also supposed to authenticate the client using client certificates.

When I try to use a proxy created from an https URI
```c#
            HttpClientHandler handler = new HttpClientHandler();
            handler.Proxy = new WebProxy("https://127.0.0.1");
            HttpClient client = new HttpClient(handler);
```
I receive an `Only the 'http' scheme is allowed for proxies.` `NotSupportedException`.

I found some indication at https://github.com/dotnet/corefx/issues/20435 that this is not supported at all, so I am reopening this as a feature request.

Thanks

### PR DESCRIPTION
**Pull request by wfurt**
fixes #31113 .
I could not find functional public proxy for testing. Plain HTTP is no brainer, HTTPS over HTTPS is somewhat more complicated and I'll ask community for validation.

### CODE REVIEW

File: src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs [hunk 1]
```diff
@@ -1525,10 +1532,18 @@ private async ValueTask<(Stream, TransportContext?)> ConnectAsync(HttpRequestMes
                 case HttpConnectionKind.ProxyConnect:
                     Debug.Assert(_originAuthority != null);
                     stream = await ConnectToTcpHostAsync(_originAuthority.IdnHost, _originAuthority.Port, request, async, cancellationToken).ConfigureAwait(false);
+                    if (_kind == HttpConnectionKind.ProxyConnect && _sslOptionsProxy != null)
```
Comment (stephentoub) [inline src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs:1535]:
```suggestion
                    if (_sslOptionsProxy != null)
```

**Review by stephentoub — APPROVED** [review]
LGTM. Thanks.

**Comment by wfurt** [comment]
failures may be relevant. Investigating..

File: src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpUtilities.cs [hunk 2]
```diff
@@ -22,7 +22,7 @@ internal static class HttpUtilities
             string.Equals(scheme, "wss", StringComparison.OrdinalIgnoreCase);
 
         internal static bool IsSupportedProxyScheme(string scheme) =>
-            string.Equals(scheme, "http", StringComparison.OrdinalIgnoreCase) || IsSocksScheme(scheme);
+            string.Equals(scheme, "http", StringComparison.OrdinalIgnoreCase) || string.Equals(scheme, "https", StringComparison.OrdinalIgnoreCase) || IsSocksScheme(scheme);
```
Comment (MihaZupan) [inline src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpUtilities.cs:25]: Likely worth updating the exception message as well
https://github.com/dotnet/runtime/blob/c8efa1204de234d8780ae993612b5f005e397651/src/libraries/System.Net.Http/src/Resources/Strings.resx#L372-L373

**Review by karelz — APPROVED** [review]

**Comment by wfurt** [comment]
> > I could not find functional public proxy for testing.
> 
> Have you considered running mitmproxy locally? We could even configure it in our enterprise tests to have proper E2E automated testing.

I can look into it. I originally wanted come public proxy on Internet.

**Pull request merged**